secatest/authorization_v1_steps.go

```diff
@@ -21,26 +20,28 @@ func (suite *testSuite) createOrUpdateRoleV1Step(
 	resource *schema.Role,
 	expectedMeta *schema.GlobalTenantResourceMetadata,
 	expectedSpec *schema.RoleSpec,
-	expectedStatusState string,
+	expectedState schema.ResourceState,
 ) {
-	t.WithNewStep(stepName, func(sCtx provider.StepCtx) {
-		suite.setAuthorizationV1StepParams(sCtx, "CreateOrUpdateRole")
-
-		resp, err := api.CreateOrUpdateRole(ctx, resource)
-		requireNoError(sCtx, err)
-		requireNotNilResponse(sCtx, resp)
-
-		if expectedMeta != nil {
-			expectedMeta.Verb = http.MethodPut
-			suite.verifyGlobalTenantResourceMetadataStep(sCtx, expectedMeta, resp.Metadata)
-		}
-
-		if expectedSpec != nil {
-			suite.verifyRoleSpecStep(sCtx, expectedSpec, &resp.Spec)
-		}
-
-		suite.verifyStatusStep(sCtx, *secalib.SetResourceState(expectedStatusState), *resp.Status.State)
-	})
+	expectedMeta.Verb = http.MethodPut
+	createOrUpdateResourceStep(
+		t,
+		ctx,
+		suite,
+		stepName,
+		suite.setAuthorizationV1StepParams,
+		"CreateOrUpdateRole",
+		resource,
+		func(context.Context, *schema.Role) (*stepFuncResponse[schema.Role, schema.GlobalTenantResourceMetadata, schema.RoleSpec], error) {
+			resp, err := api.CreateOrUpdateRole(ctx, resource)
+			return newStepFuncResponse(resp, resp.Labels, resp.Metadata, resp.Spec, resp.Status.State), err
+		},
+		nil,
+		expectedMeta,
+		suite.verifyGlobalTenantResourceMetadataStep,
+		expectedSpec,
+		suite.verifyRoleSpecStep,
+		expectedState,
+	)
```

Comment:
functions with this many arguments are hard to read and thus are a bad practice, usually no more then 4 arguments should be used for that matter. This is when argument structures come into play, instead for stashing the arguments create a structure for them, readability increases and you can also improve the code much easier.

```suggestion
	createOrUpdateResourceStep(createOrUpdateResourceOptions{
		T: t,
		ctx, //needed? why not t.Context() ?
		Suite: suite,
		Name: stepName,
		Param: suite.setAuthorizationV1StepParams,
		.... "CreateOrUpdateRole",
		resource,
		func(context.Context, *schema.Role) (*stepFuncResponse[schema.Role, schema.GlobalTenantResourceMetadata, schema.RoleSpec], error) {
			resp, err := api.CreateOrUpdateRole(ctx, resource)
			return newStepFuncResponse(resp, resp.Labels, resp.Metadata, resp.Spec, resp.Status.State), err
		},
		nil,
		expectedMeta,
		suite.verifyGlobalTenantResourceMetadataStep,
		expectedSpec,
		suite.verifyRoleSpecStep,
		expectedState,
	})
```